src/allocator/allocator.c

```
@@ -265,14 +265,32 @@ int free_raw_async(CUdeviceptr dptr, CUstream hStream) {
+static void free_unlisted_entry(allocated_list_entry *e) {
+    free(e->entry->allocHandle);
+    free(e->entry);
+    free(e);
+}
+
+/* Free the GPU memory behind e's still-unlisted allocation, then free e's
+ * heap allocations. Used when a driver call made after a successful
+ * cuMemAllocAsync/cuMemAllocFromPoolAsync fails: since e was never added to
+ * device_allocasync, the allocation would otherwise leak. */
+static void free_unlisted_alloc(allocated_list_entry *e, CUstream hStream) {
+    CUDA_OVERRIDE_CALL(cuda_library_entry, cuMemFreeAsync, e->entry->address, hStream);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)